client/src/views/LampPage.vue

```diff
@@ -126,21 +126,31 @@ function onNewAnimation() {
     })
     .then(data => {
       options.array = data.options;
-      characteristics.array = options.array.map(o => {
+      characteristics.array.splice(0, characteristics.array.length);
+      for (let i = 0; i < options.array.length; i++) {
+        const o = options.array[i];
         switch (o.type) {
           case "number":
-            return { type: "number", value: o.currentCharacteristicValue };
+            characteristics.array.push({ type: o.type, value: o.currentCharacteristicValue });
+            break;
           case "select":
-            return { type: "select", value: o.currentCharacteristicValue };
+            characteristics.array.push({ type: o.type, value: o.currentCharacteristicValue });
+            break;
           case "color":
-            return { type: "color", value: o.currentCharacteristicValue };
+            console.log(i, { type: o.type, value: o.currentCharacteristicValue });
+            characteristics.array.push({ type: o.type, value: o.currentCharacteristicValue });
+            break;
         }
-      });
+      }
     });
+}
 
+function applyAnimation() {
   axiosInstance.post("/lamp/animation", {
     animation: currentConfig.selectedAnimation,
   });
+
+  axiosInstance.post("/lamp/characteristics", characteristics.array);
```

Comment:
Would be better if it was a single endpoint in the future

Reply:
you want to be able to change the animation without reseting/re-setting the parameters.